crates/code_generation/src/node.rs

```diff
@@ -187,10 +189,20 @@ impl Node<'_> {
                         quote! { #cycler_module_name_identifier::CyclerInstance:: },
                         &self.cycler_instances.modules_to_instances[&self.node.cycler_module],
                     );
-                    // TODO: is_subscribed
+                    let path_string = path_to_path_string_token_stream(
+                        path,
+                        quote! { self.instance },
+                        quote! { #cycler_module_name_identifier::CyclerInstance:: },
+                        &self.cycler_instances.modules_to_instances[&self.node.cycler_module],
+                    );
+                    let path_string_with_dot_suffix = format!("{}.", path_string);
                     Ok(quote! {
                         #name: framework::AdditionalOutput::new(
-                            false,
+                            own_subscribed_outputs
+                                .iter()
+                                .any(|subscribed_output|
+                                    subscribed_output == #path_string || subscribed_output.starts_with(#path_string_with_dot_suffix)
+                                ),
```

Comment:
Have a look at the `should_be_filled()` function and its corresponding test case. I think, these are the relevant cases, right?

Reply:
It does cover what I remarked, yes.
But I think it regresses in another way: Previously you checked for equality or `starts_with("{path}.")`.
So if we have a hierarchy like this:
```rs
struct MainOutputs {
  a_b: AB,
  a_b_c: ABC,
}
```
And a client subscribes either of them, both will be produced, because one is equal to the subscribed path and the other is either a superstring or a substring.

Reply:
Ok, so we still need to check for equality and `starts_with` with a dot suffix? But this time with both path combinations?

Reply:
I think so, yes.

Reply:
Done, test-driven development is fun!

Reply:
Further testing has revealed, that this test still fails because the `subscribed_output` still contains the `additional_outputs.` prefix, whereas the `additional_output_path` does not.

For example, I subscrobe `additional_outputs.ball_candidates` and printed both parameters of the `should_be_filled` function:
```
additional_outputs.ball_candidates vs. image_segmenter_cycle_time
additional_outputs.ball_candidates vs. field_border_points
additional_outputs.ball_candidates vs. lines_in_image
additional_outputs.ball_candidates vs. ball_candidates
additional_outputs.ball_candidates vs. robot_detection.cluster_cones
additional_outputs.ball_candidates vs. robot_detection.cluster_points_in_pixel
additional_outputs.ball_candidates vs. robot_detection.clustered_cluster_points_in_ground
```

Reply:
Nu aber :wink: